test/xunit-runner/XUnitRunner.targets

```
@@ -54,7 +54,7 @@
       <_CurrentRuntimeTargetFramework Condition="'$(_CurrentRuntimeTargetFramework)' == ''">$(SDKCustomXUnitRuntimeTargetFramework)</_CurrentRuntimeTargetFramework>
       <_CurrentAdditionalProperties>%(SDKCustomXUnitProject.AdditionalProperties)</_CurrentAdditionalProperties>
     </PropertyGroup>
-    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);$(_CurrentAdditionalProperties)">
+    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);BuildTestPackages=false;$(_CurrentAdditionalProperties)">
```

I'd consider making this private:
```suggestion
    <MSBuild Projects="$(_CurrentSDKCustomXUnitProject)" Targets="PublishWithOutput" Properties="CustomAfterMicrosoftCommonTargets=$(_SDKCustomXUnitPublishTargetsPath);TargetFramework=$(_CurrentPublishTargetFramework);_BuildTestPackages=false;$(_CurrentAdditionalProperties)">
```

---

Isn't underscore meant for locals? I thought it indicated it was a property only local to a particular file, similar to how underscore stuff is used in other languages (meaning, other languages use it to indicate it only exists in the current 'scope').

---

I've tried searching for any, and I mean ***ANY***, documentation around MSBuild naming conventions but have never found anything formal.

---

https://learn.microsoft.com/visualstudio/msbuild/msbuild-properties#msbuild-internal-properties

It means "if you aren't us, don't use it"

---

There's nothing formal, but that will change once we implement a 'scoping' BuildCheck. There is no proper scope today as you've noted, `_` just signals 'intended to be private, please do not rely on this'. If users do so, then any breakage is on them.

---

> [learn.microsoft.com/visualstudio/msbuild/msbuild-properties](https://learn.microsoft.com/visualstudio/msbuild/msbuild-properties#msbuild-internal-properties)
> 
> It means "if you aren't us, don't use it"

Reading the doc, it actually says, "If you aren't MSBuild, don't use it." And we're not MSBuild. What I find crazy is searching for "MSBuild underscore property names" on the internet doesn't ever result in this document. I've also tried "MSBuild property naming conventions" and other such searches.

So, the document you linked further reinforces that we don't need the underscore since the interaction of this property is only scoped to our own build process and won't be exposed outside of that. It sounds like the underscore is intended for MSBuild directly to use for properties that will exist for users other than MSBuild, right? That's how I'm interpreting it. Meaning, we should never use underscores unless it is "public API surface", AKA this stuff goes to customer's build processes. This definitely does not fall into that scenario.

---

> We should update the doc - it probably hasn't been touched in quite some time. The semantic the MSbuild team communicates to all MSBuild users (internal partners, external MSBuild plugin developers, etc) is that you should not have any dependency on _any_ `_`-prefixed property, item or target to the broadest possible extent to ensure maximum compatibility.

That makes sense. So, does that mean `_`-prefixed properties are only applicable for externally exposed properties? Meaning, the intent it to avoid *other users* from using them? That would also mean that there's no purpose to use `_`-prefixed properties for build processes that are self-contained (like this one to build our Helix assets).

---

Yeah, I don't think the _ is required in this case. That's for properties we plan on shipping in the public SDK that customers consume where customers should not touch it (and yes, msbuild should build a namespacing functionality at some point).

In this case, it's a property for our own build so not being _ makes sense to me.

---

It isn't a big deal in this case, but whether it's externally visible or not isn't really the point in my mind; it's about public vs. private, and this is private. I would argue that we should never mark anything `private` in C# if we follow the externally visible rule, since we can always mark things `internal`, and that's the same from the customer's point of view. Regardless, not something I'm pushing on; I think it's fine without the underscore.

---

AFAIK, `_` isn't private, it is more "Microsoft reserved". No one outside of us uses underscores. It doesn't have any direct correlation to access control in programming languages. It is actually closer to the double underscore for identifier names in C#: https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions

> Identifiers shouldn't contain two consecutive underscore (_) characters. Those names are reserved for compiler-generated identifiers.

So, this is what I think it is supposed to be. But the confusion is that there is no "real" definition of the convention; it seems everyone interprets a convention. It could be access control or Microsoft-reserved or whatever. I just don't think it is consistently used across all of our dotnet repos. Because it was never established as a well-defined rule (or has any enforcement whatsoever).